pdata/pmetric/alias.go

```diff
@@ -74,10 +74,42 @@ const (
 )
 
 // MetricValueType specifies the type of NumberDataPoint.
+// Deprecated: [v0.49.0] Use NumberDataPointValueType or ExemplarValueType instead.
+// NOTE: It's important to choose correct type for NumberDataPoint or Exemplar, otherwise
+// it will cause a breaking change once the backward compatibility is removed.
 type MetricValueType = internal.MetricValueType
 
 const (
-	MetricValueTypeNone   = internal.MetricValueTypeNone
-	MetricValueTypeInt    = internal.MetricValueTypeInt
+	// Deprecated: [v0.49.0] Use NumberDataPointValueTypeNone or ExemplarValueTypeNone instead.
+	// NOTE: It's important to choose correct type for NumberDataPoint or Exemplar, otherwise
+	// it will cause a breaking change once the backward compatibility is removed.
+	MetricValueTypeNone = internal.MetricValueTypeNone
+
+	// Deprecated: [v0.49.0] Use NumberDataPointValueTypeInt or ExemplarValueTypeInt instead.
+	// NOTE: It's important to choose correct type for NumberDataPoint or Exemplar, otherwise
+	// it will cause a breaking change once the backward compatibility is removed.
+	MetricValueTypeInt = internal.MetricValueTypeInt
+
+	// Deprecated: [v0.49.0] Use NumberDataPointValueTypeDouble or ExemplarValueTypeDouble instead.
+	// NOTE: It's important to choose correct type for NumberDataPoint or Exemplar, otherwise
+	// it will cause a breaking change once the backward compatibility is removed.
 	MetricValueTypeDouble = internal.MetricValueTypeDouble
 )
+
+// NumberDataPointValueType specifies the type of NumberDataPoint value.
+type NumberDataPointValueType = internal.MetricValueType
+
+const (
+	NumberDataPointValueTypeNone   = internal.MetricValueTypeNone
+	NumberDataPointValueTypeInt    = internal.MetricValueTypeInt
+	NumberDataPointValueTypeDouble = internal.MetricValueTypeDouble
+)
+
+// ExemplarValueType specifies the type of Exemplar measurement value.
+type ExemplarValueType = internal.MetricValueType
+
+const (
+	ExemplarValueTypeNone   = internal.MetricValueTypeNone
+	ExemplarValueTypeInt    = internal.MetricValueTypeInt
+	ExemplarValueTypeDouble = internal.MetricValueTypeDouble
+)
```

Comment:
I don't think this is that used, so I would propose an exception here and do the correct thing, as long as we can build successfully the contrib. Give a try :)

Reply:
My idea is to see if we do the right thing for exemplar (no backwards compatible change) is ok.

Reply:
Makes sense. We can probably break `MetricValueType` -> `ExemplarValueType`, but keep `MetricValueType` <-> `NumberDataPOintValueType` backward compatible 👍

Reply:
Updated PR. Looks like Exemplar types are used at in `pkg/translator/opencensus/metrics_to_oc.go`  I don't see any other errors